lib/spack/docs/monitoring.rst

```diff
@@ -0,0 +1,88 @@
+.. Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+   Spack Project Developers. See the top-level COPYRIGHT file for details.
+
+   SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+.. _monitoring:
+
+==========
+Monitoring
+==========
+
+You can use a `spack monitor <https://github.com/spack/spack-monitor>`_ "Spackmon"
+server to store a database of your packages, builds, and associated metadata 
+for provenance, research, or some other kind of development. You should
+follow the instructions in the `spack monitor documentation <https://spack-monitor.readthedocs.org>`_
+to first create a server along with a username and token for yourself.
+You can then use this guide to interact with the server.
+
+---------------------
+Monitoring An Install
+---------------------
+
+Since an install is typically when you build packages, we logically want
+to tell spack to monitor during this step. Let's start with an example
+where we want to monitor the install of hdf5. Unless you have disabled authentication
+for the server, we first want to export our spack monitor token and username to the environment:
+
+.. code-block:: console
+ 
+    $ export SPACKMON_TOKEN=50445263afd8f67e59bd79bff597836ee6c05438
+    $ export SPACKMON_USER=spacky
+
+
+By default, the host for your server is expected to be at ``http://127.0.0.1``
+with a prefix of ``ms1``, and if this is the case, you can simply add the
+``--monitor`` flag to the install command:
+
+.. code-block:: console
+
+    $ spack install --monitor hdf5
+
+
+If you need to customize the host or the prefix, you can do that as well:
+
+.. code-block:: console
+
+    $ spack install --monitor --monitor-prefix monitor --monitor-host https://monitor-service.io hdf5
+
+
+As a precaution, we cut out early in the spack client if you have not provided
+authentication credentials. For example, if you run the command above without
+exporting your username or token, you'll see:
+
+.. code-block:: console
+
+    ==> Error: You are required to export SPACKMON_TOKEN and SPACKMON_USER
+
+This extra check is to ensure that we don't start any builds,
+and then discover that you forgot to export your token. However, if 
+your monitoring server has authentication disabled, you can tell this to
+the client to skip this step:
```

Comment:
It checks for the environment variables, and pings the monitor endpoint to check that the server is running. Since the server will ask for authentication when it's warranted (e.g., the information endpoint doesn't need auth, and the server can choose to customize which endpoints have it) we wouldn't exit with an authentication failure message until the first ping to the server, which is right after the monitor instance is created when we try to create a new Build, [here](https://github.com/spack/spack/pull/21930/files#diff-5c111ba1f5bfb321d12a190c2a79b33c7b53e3d02f6d0201f1703432b71cb17cR406). It will exit on error if the authentication fails, which is what I think we'd want if the user provides credentials that they expect to work. If the user wants to say "Try the endpoint and I don't care if it works or not, keep going) then they can use the flag `--monitor-keep-going` and it will just [keep going](https://github.com/spack/spack/pull/21930/files#diff-7f0454509928f4f8fee4c25eba560f8bc4f7a5fbc6aeef8a31075041a6df4133R215). The user should generally know if their spack monitor server requires auth, and if it does, be able to get tokens for it.

Reply:
Ok, works for me, thanks!